src/halmos/__main__.py

```diff
@@ -1340,22 +1339,18 @@ def resolve_target_contracts(ctx: InvariantTestingContext, ex: Exec) -> set[Addr
     resolved_target_contracts -= ctx.excluded_contracts
     resolved_target_contracts |= target_selectors.keys()
 
-    # Note: FOUNDRY_TEST is excluded unless a targetSelector() is specified for it, even if targetContract(FOUNDRY_TEST) is provided.
-    result = (
+    # Note: FOUNDRY_TEST is excluded unless a targetSelector() is specified for it
+    # or targetContract(FOUNDRY_TEST) is provided.
+    resolved_target_contracts = (
         resolved_target_contracts
-        if target_selectors.get(FOUNDRY_TEST)
+        if (target_selectors.get(FOUNDRY_TEST) or FOUNDRY_TEST in target_contracts)
```

Comment:
this should be faster in most cases, as it's uncommon to specify selectors without including in the target:
```suggestion
        if (FOUNDRY_TEST in target_contracts or target_selectors.get(FOUNDRY_TEST))
```